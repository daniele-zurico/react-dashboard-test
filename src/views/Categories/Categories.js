import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
import {ResponsiveDialog, Tags} from "../../components";
import {Footer} from "./categories.style";
import {MARGIN, Row} from "../../components/styles";
import TextField from "@material-ui/core/TextField/TextField";

class Categories extends Component {
  constructor(props) {
    super(props);
    this.state = {
      openDialog: false,
      categoryName: '',
      tagId: -1,
      tags: [
        {id: 1, color: '#e91e63', selected: false},
        {id: 2, color: '#2196f3', selected: false},
        {id: 3, color: '#8bc34a', selected: false},
        {id: 4, color: '#ff5722', selected: false},
      ]
    };
  }

  handleToggleDialog = () => {
    this.setState((prevState) => ({
      openDialog: !prevState.openDialog
    }));
  };

  handleConfirmDialog = () => {
    const { categoryName, tagId } = this.state;
    const payload = {
      categoryName,
      tagId
    };
    // send to graphQL
    console.log(payload);
    this.handleToggleDialog();
  };

  handleChange = name => event => {
    this.setState({ [name]: event.target.value });
  };

  handleSelectedTag = evt => {
    const id = parseInt(evt.target.id, 10);
    this.setState((prevState) => ({
      tagId: id,
      tags: prevState.tags.map((tag) => ({
        ...tag,
        selected: id === tag.id
      }))
    }));
  };

  render() {
    const {openDialog, tagId, categoryName, tags} = this.state;
    const formDisabled = tagId === -1 || categoryName === '';
    return (
      <div>
        <ResponsiveDialog
          open={openDialog}
          onConfirm={this.handleConfirmDialog}
          onCancel={this.handleToggleDialog}
          disabled={formDisabled}>
          <form noValidate autoComplete="off">
            <TextField
              required
              fullWidth
              id="categoryName"
              label="Category name"
              margin="normal"
              value = {categoryName}
              onChange={this.handleChange('categoryName')}/>
            <Row marginTop={MARGIN.TOP}>
              <div>Tags:</div>
              <Tags tags={tags} onSelected={this.handleSelectedTag}/>
            </Row>
          </form>
        </ResponsiveDialog>
        <Footer>
          <Button
            variant="fab"
            color="primary"
            aria-label="Add"
            onClick={this.handleToggleDialog}>
            <AddIcon/>
          </Button>
        </Footer>
      </div>
    );
  }
}
export { Categories };
