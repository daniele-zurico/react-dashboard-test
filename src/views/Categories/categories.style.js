import styled from "styled-components";

const Footer = styled.div`
  position: absolute;
  bottom: 10px;
  right: 10px;
  display: flex;
`;

const Label = styled.label`
  color: rgba(0, 0, 0, 0.54);
  padding: 0;
  font-size: 1rem;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  line-height: 1.7;
`;

export {Footer, Label};