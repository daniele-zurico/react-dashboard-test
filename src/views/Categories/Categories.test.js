import React from 'React';
import ReactDOM from 'react-dom';
import { Categories } from './Categories';
import { configure, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Categories/>', () => {
  const handleToggleDialog = jest.fn();
  const handleConfirmDialog = jest.fn();
  const handleChange = jest.fn();
  const handleSelectedTag = jest.fn();

  it('should render the page', () => {
    const wrapper = shallow(<Categories />);
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<Categories />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleToggleDialog once', () => {
    const wrapper = shallow(<Categories />);
    console.log(wrapper.debug());
  });
});
