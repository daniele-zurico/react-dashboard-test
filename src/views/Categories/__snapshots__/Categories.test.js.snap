// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories/> should match the snapshot 1`] = `
<div>
  <WithTheme(WithWidth(WithMobileDialog))
    disabled={true}
    onCancel={[Function]}
    onConfirm={[Function]}
    open={false}
  >
    <form
      autoComplete="off"
      noValidate={true}
    >
      <TextField
        fullWidth={true}
        id="categoryName"
        label="Category name"
        margin="normal"
        onChange={[Function]}
        required={true}
        select={false}
        value=""
      />
      <Row
        marginTop="10px"
      >
        <div>
          Tags:
        </div>
        <Tags
          onSelected={[Function]}
          tags={
            Array [
              Object {
                "color": "#e91e63",
                "id": 1,
                "selected": false,
              },
              Object {
                "color": "#2196f3",
                "id": 2,
                "selected": false,
              },
              Object {
                "color": "#8bc34a",
                "id": 3,
                "selected": false,
              },
              Object {
                "color": "#ff5722",
                "id": 4,
                "selected": false,
              },
            ]
          }
        />
      </Row>
    </form>
  </WithTheme(WithWidth(WithMobileDialog))>
  <styled.div>
    <WithStyles(Button)
      aria-label="Add"
      color="primary"
      onClick={[Function]}
      variant="fab"
    >
      <pure(Add) />
    </WithStyles(Button)>
  </styled.div>
</div>
`;
